src/Provider/PaymentRedirectionProvider.php

```diff
@@ -25,30 +25,32 @@
 
 use Configuration;
 use Context;
+use Invertus\SaferPay\Adapter\LegacyContext;
 use Invertus\SaferPay\Config\SaferPayConfig;
 use Invertus\SaferPay\Enum\ControllerName;
 use Invertus\SaferPay\Enum\PaymentType;
+use Invertus\SaferPay\Factory\ModuleFactory;
 use Invertus\SaferPay\Repository\SaferPayFieldRepository;
 
 class PaymentRedirectionProvider
 {
     /**
-     * @var Context
+     * @var LegacyContext
      */
     private $context;
 
     /**
-     * @var string
+     * @var ModuleFactory
      */
-    private $moduleName;
+    private $module;
 
     /** @var PaymentTypeProvider */
     private $paymentTypeProvider;
 
-    public function __construct(Context $context, $moduleName, PaymentTypeProvider $paymentTypeProvider)
+    public function __construct(LegacyContext $context, ModuleFactory $moduleFactory, PaymentTypeProvider $paymentTypeProvider)
```

Comment:
If it's only using the name for links, might as well just add 'saferpayofficial'. It shouldn't change anyways.

Reply:
Will not comment on all occurences, just this one, fix accordingly on other files.